tidb-latency-formula.md

```diff
@@ -0,0 +1,531 @@
+# TiDB Latency Formula
+
+This document breaks down the latency into metrics, for those who have interests in observation of the metrics, it's a guide of TiDB's critical path diagnosis.
+
+Generally, OLTP workload can be divided into read and write queries, they share some critical code.
+This document will break down the latency and then analyze it from user's perspective. It's better to read [Performance Analysis and Tuning](/performance-tuning-methods.md) before this document.
+Note when breaking down latency through metrics in this document, we are calculating the average duration or latency instead of some specific slow queries.
+Many metrics are collected as histogram, which is a distribution of the duration or latency, when calculating the average latency, you need to use the sum and count counter.
+
+```
+avg = ${metric_name}_sum / ${metric_name}_count
+```
+
+Metrics in this document can be read directly from prometheus of TiDB.
```

Comment:
Maybe we should migrate the [rendered performance map](https://you06.github.io/performance-map/) somewhere and link it here?

Reply:
It will be much better if we can put the diagrams directly on this page. They are much more readable than the formulas.
 
TiDB doc itself supports rendering railroad diagrams but click and jump may not work.

Reply:
Do you mean attaching the diagrams to the formulas instead of replacing formulas?

Reply:
> TiDB doc itself supports rendering railroad diagrams but click and jump may not work.

I made a PR to support rendering railroad diagram pingcap/website-docs#334